.toys/.toys.rb

```diff
@@ -135,6 +135,17 @@ def run
   include :exec, e: true
   include :terminal
 
+  Signal.trap("SIGINT") do
```

Comment:
Can you put this in the implementation of `toys gen` itself so it won't affect any other tools? (e.g. suppose I kick off a `toys gen` in one terminal, so there are a bunch of generation processes running in the background, and then concurrently I run a totally unrelated toys tool in another terminal, and then interrupt it. Then this trap will kick in during the unrelated run, and clobber the generation.)

One way to do this is, rather than trapping the signal globally here, add an interrupt handler on the `gen` tool implementation, which is located in `shared/.toys.rb`. Use the [`on_interrupt` directive](https://dazuma.github.io/toys/gems/toys/latest/Toys/DSL/Tool.html#on_interrupt-instance_method).

Reply:
Thanks, I moved it to live inside `gen` tool.

It looks like `on_interrupt` works well, an example of process that hangs afterwards is `ruby /usr/.../.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/bin/protoc-gen-ruby_cloud`, so I am actually using regex `/protoc-gen-ruby_/` as a match.

**Repro steps:**
1. Change `gapic-generator/templates/default/service/rest/test/method/_normal.erb` to reference an incorrect value (e.g. `s/field.name/field.wrong_name`)
2. `toys gen language_v1`.
3. Once process hangs, send interrupt signal.